### DISCUSSION TITLE
Does DeepSpeed of Pytorch-Lightning support BFloat16 now?

### DISCUSSION THREAD

**Discussion by ShaneTian** [Lightning-AI/pytorch-lightning#12481] [DDP / multi-GPU / multi-node]
How to configure BFloat16 in DeepSpeed of Pytorch-Lightning?
1. Pass to Trainer **only**: `Trainer(..., precision="bf16", ...)`
2. Configure in DeepSpeed config **only**:
    ```json
    {
      "bf16": {
        "enabled": true
      }
    }
    ```
3. **Both** setup Trainer and DeepSpeed config

---

One more thing:
I did not find anything about `bf16` in deepspeed, is it not supported now?
https://github.com/PyTorchLightning/pytorch-lightning/blob/2e5728a4841cb1d3b75123ac941cd36f681f9a11/pytorch_lightning/strategies/deepspeed.py#L652-L667

**Comment by rohitgr7** [comment]
will be added:
PR: [Add bfloat16 support to DeepSpeedStrategy](https://github.com/PyTorchLightning/pytorch-lightning/pull/12508#)
